[WIP] GHA workflows added

gzigzigzeo opened the pull request.

gzigzigzeo committed:
GHA workflows added

gzigzigzeo committed:
Merge branch 'gravitational:master' into master

gzigzigzeo committed:
Merge branch 'gravitational:master' into master

gzigzigzeo committed:
Increase timeout

gzigzigzeo committed:
Merge branch 'master' of github.com:gzigzigzeo/teleport

zmb3 commented on .cloudbuild/scripts/cmd/integration-tests/args.go:
```diff
@@ -90,6 +91,7 @@ func parseCommandLine() (*commandlineArgs, error) {
 	flag.Var(&args.artifactSearchPatterns, "a", "Path to artifacts. May be shell-globbed, and have multiple entries.")
 	flag.StringVar(&args.githubKeySrc, "key-secret", "", "Location of github deploy token, as a Google Cloud Secret")
 	flag.BoolVar(&args.skipUnshallow, "skip-unshallow", false, "Skip unshallowing the repository.")
+	flag.BoolVar(&args.forceRun, "force", false, "Run tests regardless of changes detected (ex: push to master)")
```
Curious - why do we need this?

zmb3 commented on .github/workflows/integration-tests.yaml:
```diff
@@ -0,0 +1,51 @@
+    runs-on: ubuntu-latest
+    # timeout-minutes: 10 TODO: Uncomment on production runner
+    container: 
+      image: public.ecr.aws/gravitational/teleport-buildbox:teleport11
```
Should this be `teleport12` for master?

zmb3 commented on .github/workflows/integration-tests.yaml:
```diff
@@ -0,0 +1,51 @@
+        - ${{ github.workspace }}:/workspace
+      options: --cap-add=SYS_ADMIN --privileged
+    steps:
+      - name: Install git 1.18+
```
git isn't available by default after running the checkout action?

zmb3 commented on .github/workflows/integration-tests.yaml:
```diff
@@ -0,0 +1,51 @@
+      - name: Rust cargo cache
+        uses: actions/cache@v3
+        with:
+          path: /workspace/target
```
This might cache the build as well as the dependencies - is that what we want?

zmb3 commented on .github/workflows/integration-tests.yaml:
```diff
@@ -0,0 +1,51 @@
+        if: ${{ github.ref_name == 'master' }}
+        run: |
+          cd /workspace/.cloudbuild/scripts
+          go run ./cmd/integration-tests -target "$GITHUB_REF_NAME" -commit $GITHUB_SHA -force
```
Alternatively, consider checking for the `GITHUB_ACTIONS` environment variable, and if set, then `-target` and `-commit` can be read directly from the GHA environment.

gzigzigzeo committed:
Merge branch 'gravitational:master' into master

tigrato committed:
Fixes TestListKube when running in Github Actions

When running tests in Github Actions, Teleport automatically imports the instance labels as service labels.
This results in the TestListKube test to fail since it receives unexpected labels.
This PR replaces the default IMDSClient with a fake one to avoid introducing labels imported from the CI env where the test is running.
It also removes an unrelated error message - self subject access review failed - that is printed but had no impact on the test result.

gzigzigzeo committed:
Merge remote-tracking branch 't/tigrato/fix-kube-list-auto-imported-labels'

gzigzigzeo committed:
No helm test force

gzigzigzeo committed:
Merge branch 'master' of github.com:gzigzigzeo/teleport

gzigzigzeo committed:
helm --help

gzigzigzeo committed:
Hp

gzigzigzeo committed:
helm version

gzigzigzeo committed:
helm again

gzigzigzeo committed:
Restart

gzigzigzeo committed:
Updates to unit cache

gzigzigzeo committed:
go cache dir print

r0mant commented on .cloudbuild/scripts/cmd/unit-tests/main.go:
```diff
@@ -142,16 +144,22 @@ func run() error {
+		log.Println("Forcing test run")
+	} else {
+		log.Println("Analyzing code changes")
+		ch, err := changes.Analyze(args.workspace, args.targetBranch, args.commitSHA)
```
Won't this assignment overshadow the `ch` declared in the outer scope so nothing will ever get run?

r0mant commented on .github/workflows/integration-tests.yaml:
```diff
@@ -0,0 +1,51 @@
+        run: |
+          cd /workspace/.cloudbuild/scripts
+          go run ./cmd/integration-tests -target "$GITHUB_REF_NAME" -commit $GITHUB_SHA -force
+      - name: Run tests
```
Not sure why we have "Run tests (force)" and "Run tests" - won't this run tests twice? Or is this just for testing?

r0mant commented on .github/workflows/lint.yaml:
```diff
@@ -0,0 +1,20 @@
+    name: make lint
+    runs-on: ubuntu-latest
+    container:
+      image: public.ecr.aws/gravitational/teleport-buildbox:teleport11
```
```suggestion
      image: public.ecr.aws/gravitational/teleport-buildbox:teleport12
```

r0mant commented on .github/workflows/os-compability-test.yaml:
```diff
@@ -0,0 +1,81 @@
+    runs-on: ubuntu-latest
+    # timeout-minutes: 10 TODO: Uncomment on production runner
+    container: 
+      image: public.ecr.aws/gravitational/teleport-buildbox-centos7:teleport11
```
```suggestion
      image: public.ecr.aws/gravitational/teleport-buildbox-centos7:teleport12
```

r0mant commented on tool/tsh/kube_test.go:
```diff
@@ -73,6 +74,14 @@ func TestListKube(t *testing.T) {
 			require.NoError(t, err)
 			return len(rootClusters) >= 2
 		}),
+		withNewTeleportOption(
```
I think these kube changes were a part of Tiago's fix? I think you need to rebase your fork off of latest.

gzigzigzeo committed:
Cache check

gzigzigzeo committed:
Try share action

gzigzigzeo committed:
Uses

gzigzigzeo committed:
Update unit-tests.yaml

gzigzigzeo committed:
action

gzigzigzeo committed:
Merge

gzigzigzeo committed:
Try to provide secrets

gzigzigzeo committed:
gh run id

gzigzigzeo committed:
Checkout

gzigzigzeo committed:
Github scripts

gzigzigzeo committed:
Test artifacts

gzigzigzeo committed:
Permissions

gzigzigzeo committed:
teleport-storage SA

gzigzigzeo committed:
Skip unshallow

gzigzigzeo committed:
No unshallow

gzigzigzeo committed:
Restart unit

gzigzigzeo committed:
Safe dir

gzigzigzeo committed:
unit-tests without deploy key and unshallow

gzigzigzeo committed:
Fixes to unit-tests script
